nixos/modules/virtualisation/docker.nix

```diff
@@ -185,6 +184,11 @@ in
 
         startAt = optional cfg.autoPrune.enable cfg.autoPrune.dates;
       };
+
+      boot.kernelModules =
+        (optional (
+          cfg.storageDriver == "overlay" || cfg.storageDriver == "overlay2"
+        ) "overlay");
```

Comment:
Should this not be automatically loaded, if `mount -t overlay` is used?

Reply:
Hmm It did not work for me, I should check again.